chaoscenter/web/src/views/CreateProject/CreateProject.tsx

```diff
@@ -0,0 +1,117 @@
+import type { UseMutateFunction } from '@tanstack/react-query';
+import React from 'react';
+import { FontVariation } from '@harnessio/design-system';
+import { Layout, Container, FormInput, ButtonVariation, Text, Button } from '@harnessio/uicore';
+import { Formik, Form } from 'formik';
+import { Icon } from '@harnessio/icons';
+import * as Yup from 'yup';
+import type { CreateProjectErrorResponse, CreateProjectMutationProps, CreateProjectOkResponse } from '@api/auth';
+import { useStrings } from '@strings';
+
+interface CreateProjectViewProps {
+  createProjectMutation: UseMutateFunction<
+    CreateProjectOkResponse,
+    CreateProjectErrorResponse,
+    CreateProjectMutationProps<never>
+  >;
+  createProjectMutationLoading: boolean;
+  handleClose: () => void;
+}
+
+interface CreateProjectFormProps {
+  projectName: string;
+  projectDescription: string;
+  tags: string[];
+}
+
+export default function CreateProjectView(props: CreateProjectViewProps): React.ReactElement {
+  const { createProjectMutation, createProjectMutationLoading, handleClose } = props;
+  const { getString } = useStrings();
+
+  function handleSubmit(values: CreateProjectFormProps): void {
+    createProjectMutation(
+      {
+        body: {
+          projectName: values.projectName,
+          description: values.projectDescription,
+          tags: values.tags
+        }
+      },
+      {
+        onSuccess: () => {
+          handleClose();
+        }
+      }
+    );
+  }
+
+  const initialValues: CreateProjectFormProps = {
+    projectName: '',
+    projectDescription: '',
+    tags: []
+  };
+
+  return (
+    <Layout.Vertical padding="medium" style={{ gap: '1rem' }}>
+      <Layout.Horizontal flex={{ alignItems: 'center', justifyContent: 'space-between' }}>
+        <Text font={{ variation: FontVariation.H4 }}>{'Create Project'}</Text>
+      </Layout.Horizontal>
+      <Container>
+        <Formik<CreateProjectFormProps>
+          initialValues={initialValues}
+          onSubmit={values => handleSubmit(values)}
+          validationSchema={Yup.object().shape({
+            projectName: Yup.string().required(getString('usernameIsRequired')),
```

Comment:
We should not allow special characters except `-` or `_` IMO, if we have a BE check for this we can add it in FE also.

Reply:
hey @hrishavjha there's not validation checks for project name in backend. I would like to add them in a different pr wdyt?

Reply:
sure, sounds good to me!